dborutils/mongo_client.py

```diff
@@ -259,33 +259,39 @@ def parse_db_argstring(cls, host_spec):
 
         Returns tuple() or tuple(host, port, database)
         """
+        default_port = 27017
+        host = port = database = user_pass = None
 
-        DEFAULT_HOST = 'localhost'
-        DEFAULT_PORT = 27017
-        DEFAULT_COLLECTION = None
-
-        result = []
-
+        # Parse out host spec string into tuple(host, port, database, collection) result.
         if host_spec:
+            if re.search(r'@', host_spec):
+                user_pass, host_spec = host_spec.split('@')
 
-            result = host_spec.split(':')
+            host_spec_parts = host_spec.split(':')
 
-            if len(result) == 1:
+            if len(host_spec_parts) == 1:
                 # database
-                if not DEFAULT_HOST:
-
-                    raise Exception("Default host is required when provided a database and collection name only.")
+                raise Exception("Host is required when provided only a database name.")
 
-                result.insert(0, DEFAULT_HOST)
-                result.insert(1, DEFAULT_PORT)
-                result.insert(3, DEFAULT_COLLECTION)
-            elif len(result) == 2:
+            if len(host_spec_parts) == 2:
```

Comment:
This probably should be an `elif`, if we want to raise the exception below when there is not 1, 2, or 3 host spec parts.